test/integration/allocatable_test.go

```diff
@@ -0,0 +1,194 @@
+	imageutils "k8s.io/kubernetes/test/utils/image"
+
+	"sigs.k8s.io/scheduler-plugins/pkg/noderesources"
+	"sigs.k8s.io/scheduler-plugins/test/util"
+)
+
+func TestAllocatablePlugin(t *testing.T) {
+	registry := fwkruntime.Registry{noderesources.AllocatableName: noderesources.NewAllocatable}
+	profile := schedapi.KubeSchedulerProfile{
+		SchedulerName: v1.DefaultSchedulerName,
+		Plugins: &schedapi.Plugins{
+			Score: &schedapi.PluginSet{
+				Enabled: []schedapi.Plugin{
+					{Name: noderesources.AllocatableName,
+						Weight: 50000},
+				},
+				Disabled: []schedapi.Plugin{
+					{Name: "*"},
+				},
+			},
+		},
+	}
+
+	testCtx := util.InitTestSchedulerWithOptions(
+		t,
+		testutils.InitTestMaster(t, "sched-allocatable", nil),
+		true,
+		scheduler.WithProfiles(profile),
+		scheduler.WithFrameworkOutOfTreeRegistry(registry),
+	)
+
+	defer testutils.CleanupTest(t, testCtx)
+
+	cs, ns := testCtx.ClientSet, testCtx.NS.Name
+	// Create a Node.
+	nodeName1 := "fake-node-small-1"
+	node := st.MakeNode().Name(nodeName1).Label("node", nodeName1).Obj()
+	node.Status.Allocatable = v1.ResourceList{
+		v1.ResourcePods:   *resource.NewQuantity(32, resource.DecimalSI),
+		v1.ResourceCPU:    *resource.NewMilliQuantity(500, resource.DecimalSI),
+		v1.ResourceMemory: *resource.NewQuantity(200, resource.DecimalSI),
+	}
+	node.Status.Capacity = v1.ResourceList{
+		v1.ResourcePods:   *resource.NewQuantity(32, resource.DecimalSI),
+		v1.ResourceCPU:    *resource.NewMilliQuantity(500, resource.DecimalSI),
+		v1.ResourceMemory: *resource.NewQuantity(200, resource.DecimalSI),
+	}
+	node, err := cs.CoreV1().Nodes().Create(context.TODO(), node, metav1.CreateOptions{})
+	if err != nil {
+		t.Fatalf("Failed to create Node %q: %v", nodeName1, err)
+	}
+
+	// Create a same size node.
+	nodeName2 := "fake-node-small-2"
+	node = st.MakeNode().Name(nodeName2).Label("node", nodeName2).Obj()
+	node.Status.Allocatable = v1.ResourceList{
+		v1.ResourcePods:   *resource.NewQuantity(32, resource.DecimalSI),
+		v1.ResourceCPU:    *resource.NewMilliQuantity(500, resource.DecimalSI),
+		v1.ResourceMemory: *resource.NewQuantity(200, resource.DecimalSI),
+	}
+	node.Status.Capacity = v1.ResourceList{
+		v1.ResourcePods:   *resource.NewQuantity(32, resource.DecimalSI),
+		v1.ResourceCPU:    *resource.NewMilliQuantity(500, resource.DecimalSI),
+		v1.ResourceMemory: *resource.NewQuantity(200, resource.DecimalSI),
+	}
+	node, err = cs.CoreV1().Nodes().Create(context.TODO(), node, metav1.CreateOptions{})
+	if err != nil {
+		t.Fatalf("Failed to create Node %q: %v", nodeName2, err)
+	}
+
+	// Create a bigger Node.
+	nodeName3 := "fake-node-big"
+	node = st.MakeNode().Name(nodeName3).Label("node", nodeName3).Obj()
+	node.Status.Allocatable = v1.ResourceList{
+		v1.ResourcePods:   *resource.NewQuantity(32, resource.DecimalSI),
+		v1.ResourceCPU:    *resource.NewMilliQuantity(500, resource.DecimalSI),
+		v1.ResourceMemory: *resource.NewQuantity(5100, resource.DecimalSI),
```

Comment:
Any particular reason to make it 5100 instead of 5000?

Reply:
no, changed it to 5000